libs/angular-remote-components/src/lib/services/permission.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core'
+import { PermissionsTopic } from '@onecx/integration-interface'
+import { filter, firstValueFrom, map } from 'rxjs'
+
+@Injectable({ providedIn: 'root' })
+export class PermissionService {
+  private permissionsTopic$ = new PermissionsTopic()
+
+  async getPermissions(appId: string, productName: string): Promise<string[]> {
+    this.permissionsTopic$.publish({ appId: appId, productName: productName })
```

Comment:
You should publish after the reading because it can be the case that publish is directly triggering the response. In reality in the current implementation of the topic this will never happen because of how JS is working but we should make it independent from the current implementation